src/Libraries/Module/Templates/Demo/Api/Middlewares/Password.tpl

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * Quantum PHP Framework
+ *
+ * An open source software development framework for PHP
+ *
+ * @package Quantum
+ * @author Arman Ag. <arman.ag@softberg.org>
+ * @copyright Copyright (c) 2018 Softberg LLC (https://softberg.org)
+ * @link http://quantum.softberg.org/
+ * @since 2.9.5
+ */
+
+namespace {{MODULE_NAMESPACE}}\Middlewares;
+
+use Quantum\Libraries\Validation\Validator;
+use Quantum\Libraries\Validation\Rule;
+use Quantum\Libraries\Hasher\Hasher;
+use Quantum\Middleware\QtMiddleware;
+use Quantum\Factory\ModelFactory;
+use Quantum\Http\Response;
+use Quantum\Http\Request;
+use Shared\Models\User;
+use Closure;
+
+/**
+ * Class Password
+ * @package Modules\Web
+ */
+class Password extends QtMiddleware
+{
+    /**
+     * @var Validator
+     */
+    private $validator;
+
+    /**
+     * Class constructor
+     * @param Request $request
+     */
+    public function __construct(Request $request)
+    {
+        $this->validator = new Validator();
+        $hasher = new Hasher();
+        $hasher->setAlgorithm(PASSWORD_BCRYPT);
+        $user = ModelFactory::get(User::class)->findOneBy('uuid', $request->get('uuid', null));
+        $currentPassword = $request->get('current_password');
+
+        $this->validator->addValidation('incorrect_password', function ($value) use ($user, $hasher, $currentPassword) {
+            return $hasher->check($currentPassword, $user->password);
+        });
+
+        $this->validator->addRules([
+            'current_password' => [
+                Rule::set('required'),
+                Rule::set('incorrect_password')
+            ],
+            'new_password' => [
+                Rule::set('required'),
+                Rule::set('minLen', 6)
+            ],
+            'confirm_password' => [
+                Rule::set('required'),
+            ],
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     * @param Response $response
+     * @param Closure $next
+     */
+    public function apply(Request $request, Response $response, Closure $next)
+    {
+        if ($request->isMethod('post')) {
+            if (!$this->validator->isValid($request->all())) {
+                $response->json([
+                    'status' => 'error',
+                    'message' => $this->validator->getErrors()
+                ]);
+
+            } else if (!$this->confirmPassword($request->get('new_password'), $request->get('confirm_password'))) {
+                $response->json([
+                    'status' => 'error'
```

Comment:
Why no messages sent in response when confirm password doesn't matching to new password?